src/m365/spo/commands/site/site-remove.ts

```diff
@@ -76,259 +80,185 @@ class SpoSiteRemoveCommand extends SpoCommand {
+          await this.deleteSiteFromSharePointRecycleBin(logger, options.url);
+        }
       }
     }
     catch (err: any) {
       this.handleRejectedPromise(err);
     }
   }
 
-  private isSiteGroupDeleted(groupId: string): Promise<{ value: { id: string }[] }> {
-    const requestOptions: any = {
-      url: `https://graph.microsoft.com/v1.0/directory/deletedItems/Microsoft.Graph.Group?$select=id&$filter=groupTypes/any(c:c+eq+'Unified') and startswith(id, '${groupId}')`,
-      headers: {
-        accept: 'application/json;odata.metadata=none'
-      },
-      responseType: 'json'
-    };
-
-    return request.get<{ value: { id: string }[] }>(requestOptions);
-  }
+  private async removeGroupFromEntraRecycleBin(logger: Logger, groupId: string): Promise<void> {
+    if (this.verbose) {
+      await logger.logToStderr(`Permanently deleting group '${groupId}'.`);
+    }
 
-  private async deleteOrphanedSite(logger: Logger, url: string): Promise<void> {
-    const spoAdminUrl: string = await spo.getSpoAdminUrl(logger, this.debug);
-    const requestOptions: any = {
-      url: `${spoAdminUrl}/_api/GroupSiteManager/Delete?siteUrl='${url}'`,
+    const requestOptions: CliRequestOptions = {
+      url: `https://graph.microsoft.com/v1.0/directory/deletedItems/Microsoft.Graph.Group/${groupId}`,
       headers: {
-        'content-type': 'application/json;odata=nometadata',
-        accept: 'application/json;odata=nometadata'
-      },
-      responseType: 'json'
+        accept: 'application/json;odata.metadata=none'
+      }
     };
 
-    return await request.post(requestOptions);
+    return request.delete(requestOptions);
   }
 
-  private async deleteSiteWithoutGroup(logger: Logger, args: CommandArgs): Promise<void> {
-    this.spoAdminUrl = await spo.getSpoAdminUrl(logger, this.debug);
-    this.context = await spo.ensureFormDigest(this.spoAdminUrl, logger, this.context, this.debug);
-
-    if (args.options.fromRecycleBin) {
-      if (this.verbose) {
-        await logger.logToStderr(`Deleting site from recycle bin ${args.options.url}...`);
-      }
-
-      await this.deleteSiteFromTheRecycleBin(args.options.url, args.options.wait, logger);
-    }
-    else {
-      await this.deleteSite(args.options.url, args.options.wait, logger);
+  private async isGroupPermanentlyDeleted(logger: Logger, groupId: string): Promise<boolean> {
+    if (this.verbose) {
+      await logger.logToStderr(`Checking if group '${groupId}' is already permanently deleted from recycle bin.`);
     }
 
-    if (args.options.skipRecycleBin) {
-      if (this.verbose) {
-        await logger.logToStderr(`Also deleting site from tenant recycle bin ${args.options.url}...`);
-      }
-
-      await this.deleteSiteFromTheRecycleBin(args.options.url, args.options.wait, logger);
-    }
+    const deletedGroups = await odata.getAllItems<{ id: string }>(`https://graph.microsoft.com/v1.0/directory/deletedItems/Microsoft.Graph.Group?$select=id&$filter=id eq '${groupId}'`);
```

Comment:
Noticed that this request fails when there was no group found.
For example when you have a site in a recycle bin with a group that is permanently deleted, this request throws an error.
Maybe let's just check for a 404 response status code.